src/ussd/v1/ussd.js

```diff
@@ -0,0 +1,336 @@
+'use strict';
+require('dotenv').config();
+const randomstring = require('randomstring');
+
+const db = require('../../../database/models/index.js');
+
+const optsUssd = {
+  schema: {
+    description: 'Ussd Endpoint Route for Farmpro App',
+    tags: ['ussd'],
+    querystring: {
+      type: 'object',
+      required: ['sessionId', 'code', 'phone'],
+      properties: {
+        sessionId: { type: 'string' },
+        code: { type: 'string' },
+        phone: { type: 'string', maxLength: 12, minLength: 12 },
+        text: { type: 'string', default: '' },
+      },
+    },
+
+    response: {
+      200: {
+        description: 'Success Response',
+        content: 'text/plain',
+        type: 'object',
+      },
+      400: {
+        description: 'Error Response',
+        content: 'application/json',
+        type: 'object',
+      },
+    },
+  },
+};
+
+module.exports = function (fastify, options, done) {
+  fastify.get('/ussd', optsUssd, async (request, reply) => {
+    let isRegistered = false;
+    let message = '';
+    let registration_level = 0;
+    const sessionId = request.query.sessionId;
+    const serviceCode = request.query.code;
+    const phoneNumber = request.query.phone;
+    const text = request.query.text;
+    console.log('text-', text);
+    let txt = request.query.text.split(/[*#]/);
+    const txtlen = txt.length;
+    console.log(txtlen);
+    if (text === '') {
+      // save ussd session to Database here
+      await db.Ussds.create({
+        sessionId: sessionId,
+        serviceCode: serviceCode,
+        level: 0,
+      });
+      // check if farmer is registered
+      const farmer = await db.Farmers.findOne({
+        where: {
+          phone: phoneNumber,
+        },
+      });
+      // if not registered prompt the farmer to register
+      if (!farmer) {
+        message = 'CON Welcome to Farm Pro \n';
+        message += 'Reply with 1 to Register \n';
+        message += '1: Register\n';
+        // if registered let the farmer access the menus
+      } else {
+        const farmerName = farmer.firstName
+          .split(' ')
+          .map((w) => w[0].toUpperCase() + w.substr(1).toLowerCase())
+          .join(' ');
+        isRegistered = true;
+        message = `CON Welcome ${farmerName} to Farm Pro \n`;
+        message += 'Menu\n';
+        message += '1: My Farm\n';
+        message += '2: Learn\n';
+        message += '3: Loans\n';
+        message += '4: My Profile\n';
+      }
+    } else if (parseInt(txt[0]) === 1 && isRegistered === false) {
+      const ussd_level = await db.Ussds.findOne({
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Identifier 'ussd_level' is not in camel case.](https://app.codacy.com/gh/BuildForSDG/Farm-Pro/pullRequest?prid=5674718)